hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -113,13 +119,14 @@ private AbfsClient(final URL baseUrl, final SharedKeyCredentials sharedKeyCreden
     this.retryPolicy = abfsClientContext.getExponentialRetryPolicy();
     this.accountName = abfsConfiguration.getAccountName().substring(0, abfsConfiguration.getAccountName().indexOf(AbfsHttpConstants.DOT));
     this.authType = abfsConfiguration.getAuthType(accountName);
+    this.encryptionContextProvider = encryptionContextProvider;
 
     String encryptionKey = this.abfsConfiguration
         .getClientProvidedEncryptionKey();
     if (encryptionKey != null) {
-      this.clientProvidedEncryptionKey = getBase64EncodedString(encryptionKey);
-      this.clientProvidedEncryptionKeySHA = getBase64EncodedString(
-          getSHA256Hash(encryptionKey));
+      this.clientProvidedEncryptionKey = EncryptionAdapter.getBase64EncodedString(encryptionKey);
```

Comment:
driver needs to send two request headers:
1. Base 64 encoded key
2. Base 64 encoded (SHA256 hash of key)
As the second header requires the original key to compute the SHA256, the config needs to accept the key without the encoding

Reply:
Well, it can just accept the encoded key and decode it. I am not sure which one is the best way, I was just pointing out that azCopy requires the encoded key instead.

Reply:
azCopy takes precomputed values as configs, i.e., both the encoded key value and the encoded SHA256 hash of the key are required as input. ABFS driver currently computes these headers from the original key provided. Both of these methods are quite similar, hence would like to know your opinion on whether you see a value add in the driver change (to switch to the 2 precomputed key values for configs)

Reply:
I am advocating in favor of the encoded version of the key for the following reasons:

- it might be easier to manage base 64 encoded strings in the configuration
- in ingestion flows where users use azCopy to upload data and then use the ABFS client to access them, they would need to provide both encoded (azCopy) and unencoded (this PR) version of the key.

Alternatively, we could also use both precomputed values (key and sha) in the provided configuration for the client.

Reply:
Have modified to accept encoded version, but will require both pre-computed key and SHA hash value as configs